### ISSUE TITLE
Fails to source "logstash.lib.sh" when starting over SSH

### ISSUE THREAD

**Issue by jmcomets** [elastic/logstash#3970]
Hi, I'm trying to start logstash and detach as a SSH one-liner in a remote management script. I get the following error when running `ssh user@host cd path/to/logstash && bin/logstash`

    bin/logstash: $(cd bin/..; pwd)/bin/logstash.lib.sh: not found

When I run the command in an interactive SSH session I don't get the above error.

Please note that I'm running this on a Windows machine, and pointing to a SunOS VM. I tried using Logstash 1.5.4 and the latest 2.0.0.beta1 and encountered the same issue.

**Comment by hypergori** [comment]
Had the same issue but , after I changed the shell script to use bash on Solaris then , it started to work.

bin/logstash
bin/logstash.lib.sh

on the 1st line , I use #!/usr/bin/bash

**Comment by jordansissel** [comment]
What OS? What is your /bin/sh? Is it POSIX-compliant? I'm pretty sure `$(...)` is specified in POSIX, and that's what the shell script targets.

**Comment by aayoubi** [comment]
On Solaris 10, `/usr/bin/sh` is not a standards compliant shell. The POSIX compliant shell is located under `/usr/xpg4/bin/sh`. Would it be possible to use `bash` by default?

**Comment by jordansissel** [comment]
You could change this, but I don't know if we can change it for logstash
itself (making the path change to bash would break other operating systems)

**Comment by aayoubi** [comment]
I agree. In all cases, this only occurs on Solaris 10, I don't see this issue on Solaris 11.
